tiles/src/main/java/com/protomaps/basemap/layers/Water.java

```diff
@@ -21,262 +31,378 @@ public class Water implements ForwardingProfile.LayerPostProcessor {
 
-          // Server sort features so client label collisions are pre-sorted
-          waterLabelPosition.setSortKey((int) minZoom);
+    if (sf.canBePolygon() && minZoomString != null) {
+      int minZoom = (int) Math.round(Double.parseDouble(minZoomString));
 
-          NeNames.setNeNames(waterLabelPosition, sf, 0);
-        }
-      }
+      int themeMinZoom = sf.getSourceLayer().contains("_50m_") ? 0 : 5;
+      int themeMaxZoom = sf.getSourceLayer().contains("_50m_") ? 4 : 5;
+
+      features.polygon(LAYER_NAME)
+        .setAttr("kind", kind)
+        .setAttr("sort_rank", 200)
+        .setPixelTolerance(Earth.PIXEL_TOLERANCE)
+        .setZoomRange(Math.max(themeMinZoom, minZoom), themeMaxZoom)
+        .setMinPixelSize(1.0)
+        .setBufferPixels(8);
     }
   }
 
   public void processOsm(SourceFeature sf, FeatureCollector features) {
-    // polygons
-    if (sf.canBePolygon() && (sf.hasTag("water") ||
-      (sf.hasTag("waterway") && !sf.hasTag("waterway", "dam")) ||
-      sf.hasTag("natural", "water") ||
-      sf.hasTag("landuse", "reservoir") ||
-      sf.hasTag("leisure", "swimming_pool"))) {
-      String kind = "other";
-      String kindDetail = "";
-      var reservoir = false;
-      var alkaline = false;
-
-      // coalesce values across tags to single kind value
-      if (sf.hasTag("natural", "water", "bay", "strait", "fjord")) {
-        kind = sf.getString("natural");
-
-        if (sf.hasTag("amenity", "fountain")) {
-          kind = "fountain";
-        }
-        if (sf.hasTag("water", "basin", "canal", "ditch", "drain", "lake", "river", "stream")) {
-          kindDetail = sf.getString("water");
-
-          // This is a bug in Tilezen v1.9 that should be fixed in 2.0
-          // But isn't present in Protomaps v2 so let's fix it preemptively
-          if (kindDetail.equals("lake")) {
-            kind = "lake";
-          }
-
-          if (sf.hasTag("water", "lagoon", "oxbow", "pond", "reservoir", "wastewater")) {
-            kindDetail = "lake";
-          }
-          if (sf.hasTag("water", "reservoir")) {
-            reservoir = true;
-          }
-          if (sf.hasTag("water", "lagoon", "salt", "salt_pool")) {
-            alkaline = true;
-          }
-        }
-      } else if (sf.hasTag("waterway", "riverbank", "dock", "canal", "river", "stream", "ditch", "drain")) {
-        kind = "water";
-        kindDetail = sf.getString("waterway");
-      } else if (sf.hasTag("landuse", "basin")) {
-        kind = sf.getString("landuse");
-      } else if (sf.hasTag("landuse", "reservoir")) {
-        kind = "water";
-        kindDetail = sf.getString("landuse");
-        reservoir = true;
-      } else if (sf.hasTag("leisure", "swimming_pool")) {
-        kind = "swimming_pool";
-      } else if (sf.hasTag("amenity", "swimming_pool")) {
-        kind = "swimming_pool";
-      }
+    if (!sf.tags().isEmpty() && sf.isPoint()) {
+      sf.setTag("_is_point", "yes");
```

Comment:
should we move these internally into Matcher if we're going to use this feature in other layers?